src/main/java/org/spongepowered/common/event/tracking/context/BlockPosMultiMap.java

```
@@ -0,0 +1,268 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.event.tracking.context;
+
+import com.google.common.collect.LinkedListMultimap;
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Multimaps;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.util.math.BlockPos;
+import org.spongepowered.api.block.BlockSnapshot;
+import org.spongepowered.api.event.block.ChangeBlockEvent;
+import org.spongepowered.api.world.World;
+import org.spongepowered.common.block.BlockUtil;
+import org.spongepowered.common.block.SpongeBlockSnapshot;
+import org.spongepowered.common.block.SpongeBlockSnapshotBuilder;
+import org.spongepowered.common.event.tracking.PhaseTracker;
+import org.spongepowered.common.event.tracking.TrackingUtil;
+import org.spongepowered.common.world.BlockChange;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import javax.annotation.Nullable;
+
+public final class BlockPosMultiMap implements ICaptureSupplier {
+
+    @Nullable private ListMultimap<BlockPos, SpongeBlockSnapshot> multimap;
+    @Nullable private List<SpongeBlockSnapshot> snapshots;
```

Based on how this field is used, it looks like you could just initialize it here - none of the other methods ascribe any special meaning to it being `null`.

---

There's plenty of meaning for it to be `null` by default:
- No blocks captured? No wasted memory
- `isEmpty()` checks are blazing fast
- `put(BlockSnapshot)` is more fasts on initialization due to fail fast if statements on nullability
- Because there's thousands of these objects being created per tick, they need to construct and be as lightweight as possible.

Remember, a lot of the capture suppliers have had nullable fields for these reasons to lower the average memory cost of the PhaseTracker in general.